model/riscv_insts_zbkx.sail

```diff
@@ -0,0 +1,39 @@
+/* ****************************************************************** */
+union clause ast = RISCV_XPERM8 : (regidx, regidx, regidx)
+
+mapping clause encdec = RISCV_XPERM8(rs2, rs1, rd) if haveZbkx()
+  <-> 0b0010100 @ rs2 @ rs1 @ 0b100 @ rd @ 0b0110011 if haveZbkx()
+
+mapping clause assembly = RISCV_XPERM8(rs2, rs1, rd)
+  <-> "xperm8" ^ spc() ^ reg_name(rd) ^ sep() ^ reg_name(rs1) ^ sep() ^ reg_name(rs2)
+
+function clause execute (RISCV_XPERM8(rs2, rs1, rd)) = {
+  let rs1_val = X(rs1);
+  let rs2_val = X(rs2);
+  result : xlenbits = zeros();
+  foreach (i from 0 to (sizeof(xlen) - 8) by 8) {
+    result[i+7..i] = (rs1_val >> (rs2_val[i+7..i] @ 0b000)[7..0])[7..0];
```

Comment:
For a model these would be better written as the
```
let index : bits(8) = rs2_val[i+7..i];
result[i+7...i] = if index < sizeof(xlen)/8
                  then rs1_val[8*index+7..8*index]
                  else zeros()
```
and equivalent for xperm4 versions, much easier to understand

Reply:
Same question for my information: better to use <<3 or *8 ?

Reply:
> let index : bits(8) = rs2_val[i+7..i];
> result[i+7...i] = if index < sizeof(xlen)/8
>                   then rs1_val[8*index+7..8*index]
>                   else zeros()

overloading for (operator <), can not unify int('n) and bitvector(8, dec)        ( if condition)

Reply:
Yeah ok `unsigned(index) < sizeof(xlen)/8`

Reply:
No val spec found for (operator /)

Reply:
I used 
    index : int = unsigned(rs2_val[i+3..i]);
    result[i+3..i] = if index < quot_round_zero(sizeof(xlen), 8)
                      then rs1_val[8*index+3..8*index]
                      else zeros()
And have following issue:
![image](https://user-images.githubusercontent.com/63648044/140597384-8b491028-5101-4c9d-bc89-3456643c9251.png)

Reply:
Ok if we don't have division then just do `8 * index < sizeof(xlen)`, that'll probably appease the range checker too

Reply:
Yeah, ```    let index = unsigned(rs2_val[i+7..i]);
    result[i+7..i] = if 8*index < sizeof(xlen)
                     then rs1_val[8*index+7..8*index]
                     else zeros()
``` works

Reply:
(And do make sure to use let, without it you have a mutable variable, which should only be done when you actually need the mutability)

Reply:
Still the issue in above issue not resolved.

Reply:
Resolved! Thanks